tests/pipeline/test_pipeline.py

```diff
@@ -949,7 +951,7 @@ def fail_extract():
     assert py_ex.value.step == "extract"
 
     os.environ["COMPLETED_PROB"] = "0.0"
-    os.environ["RAISE_ON_FAILED_JOBS"] = "true"
+    os.environ["RAISE_ON_FAILED_JOBS"] = "true"  # TODO: why is this necessary?
```

Comment:
Without setting this env, the pipeline does not raise. However, it should be the default. Thus, the implementation must be incomplete.